utils/realtime.h

```diff
@@ -48,13 +57,18 @@ inline void configure_cpu(int cpu) {
  * \throw std::runtime_error If the operation failed.
  */
 inline void configure_scheduler(int priority) {
+#ifndef __APPLE__
   constexpr int kPid = 0;  // this thread
   struct sched_param params = {};
   params.sched_priority = priority;
   if (::sched_setscheduler(kPid, SCHED_RR, &params) < 0) {
     throw std::runtime_error(
         "Error setting realtime scheduler, try running as root (use sudo)");
   }
+#else
+  spdlog::warn(
+      "This function has not been implemented for MacOS, doing nothing...");
```

Comment:
```suggestion
      "[utils::configure_scheduler] This function does nothing on macOS");
```

Writing out the function name for context.